Chabot 2.5

```diff
@@ -0,0 +1,122 @@
+    
+    recognition.onspeechstart = function () {
+        document.dispatchEvent(new CustomEvent("GET_ONREC", {detail: 'running'}));
+    }
+    recognition.onsoundend = function () {
+        document.dispatchEvent(new CustomEvent("GET_ONREC", {detail: 'stop'}));
+    }
+    recognition.onresult = function (e) {
+        var value2 = "";
+        for (var i = e.resultIndex; i < e.results.length; ++i) {
+            if (e.results[i].isFinal) {
+                value += e.results[i][0].transcript;
+                rand = Math.random();
+                
+            } else {
+                value2 += e.results[i][0].transcript;
+            }
+        }
+        document.dispatchEvent(new CustomEvent("GET_TEXT", {detail: {t:value, s:rand}}));
+        document.dispatchEvent(new CustomEvent("GET_INTRM", {detail: value2}));
+
+    }
+    recognition.onerror = function(e) {
+        document.dispatchEvent(new CustomEvent("GET_ONREC", {detail: 'stop'}));
+    }
+    recognition.start();
+    """))
+
+result = streamlit_bokeh_events(
+    bokeh_plot = stt_button,
+    events="GET_TEXT,GET_ONREC,GET_INTRM",
+    key="listen",
+    refresh_on_update=False,
+    override_height=75,
+    debounce_time=0)
+
+tr = st.empty()
+
+if 'input' not in st.session_state:
+    st.session_state['input'] = dict(text='', session=0)
+
+tr.text_area("**Your input**", value=st.session_state['input']['text'])
+
+if result:
+    if "GET_TEXT" in result:
+        if result.get("GET_TEXT")["t"] != '' and result.get("GET_TEXT")["s"] != st.session_state['input']['session'] :
+            st.session_state['input']['text'] = result.get("GET_TEXT")["t"]
+            tr.text_area("**Your input**", value=st.session_state['input']['text'])
+            st.session_state['input']['session'] = result.get("GET_TEXT")["s"]
+
+    if "GET_INTRM" in result:
+        if result.get("GET_INTRM") != '':
+            tr.text_area("**Your input**", value=st.session_state['input']['text']+' '+result.get("GET_INTRM"))
+
+    if "GET_ONREC" in result:
+        if result.get("GET_ONREC") == 'start':
+            placeholder.title("You can speak now!")
+            st.session_state['input']['text'] = ''
+        elif result.get("GET_ONREC") == 'running':
+            placeholder.title("Transcribing")
+        elif result.get("GET_ONREC") == 'stop':
+            placeholder.title("Click Again To Ask Another Question")
+            if st.session_state['input']['text'] != '':
+                input = st.session_state['input']['text']
+                output = generate_response(input)
+                st.write("**ChatBot:**")
+                st.write(output)
+                st.session_state['input']['text'] = ''
+#conda activate voice_recognition_software
+                tts = gTTS(output, lang='en', tld='com')
+                file_name = f"C:/Users/darin/OneDrive/Desktop/ChatbotJava/ChatbotJava/src/welcome_{str(uuid.uuid4())}.mp3"
```

Comment:
Change this to your file destination for the mp3, use / forward slashes for between the folders